cmake/PahoMqttC.cmake

```diff
@@ -39,6 +39,7 @@ FetchContent_MakeAvailable(paho.mqtt.c-external)
 # Set dependencies and target to link to
 if (NOT OPENSSL_OFF)
     add_library(paho.mqtt.c ALIAS paho-mqtt3cs-static)
+    add_dependencies(common_ssl_obj_static OpenSSL::SSL OpenSSL::Crypto)
```

Comment:
Isn't it enough to just link them?

Reply:
The problem was that when paho-mqtt and the ssl library were built in parallel it could happen that the ssl headers were still not present in the ssl install dir when the paho-mqtt build needed them thus the build failed. So we need to wait for the ssl library to finish building before building the paho-mqtt.